pkg/dispatcher/workers/controller.go

```diff
@@ -142,6 +143,16 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 		return fmt.Errorf("couldn't get a list of calculations: %v", err)
 	}
 
+	workerPool := &workerpool.WorkerPool{}
+	err = r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: req.Name}, workerPool)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return fmt.Errorf("failed to get worker pool: %s in namespace %s: %w", req.Name, req.Namespace, err)
+	}
+	if workerPoolState, ok := workerPool.Spec.Workers[workerpool.WorkersName]; ok {
```

Comment:
If the worker doesn't exist, we should register it.

Reply:
i really cant find anything as `register`. Is the `create` method what you mean?

https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.3/pkg/client#example-Client-Create